amaUtilities.py

```diff
@@ -110,31 +120,368 @@ def addXYDistAngle(dbData, line, point1, point2, projstr, name='event'):
+                dbData.loc[index, '%s_Line' % name] = line2
+                dbData.loc[index, '%s_LineStart' % name] = length(split(row[line], row[point1]).geoms[0])
+                elevationDrop = row[point1].z - row[point2].z
+                dbData.loc[index, '%s_Angle' % name] = np.rad2deg(np.arctan(elevationDrop / length(line2)))
+                dbData.loc[index, '%s_LineAltDrop' % name] = elevationDrop
+                # compute path segment lengths and cumulative length
+                distP = gT.computeAlongLineDistance(row['geom_path_ln3d_%s' % projstr], dim='2D')
+                # compute xyLineh segment lengths and cumulative length
+                distP2 = gT.computeAlongLineDistance(line2, dim='2D')
+                # append to list for each event (row) in DF
+                distancePath.append(distP)
+                distanceEvent.append(distP2)
+                
+        else:
+          dbData.loc[index, '%s_Distance' %name] = np.nan
+          dbData.loc[index, '%s_Line' %name] = np.nan
+          dbData.loc[index, '%s_LineStart' %name] = np.nan
+          dbData.loc[index, '%s_Angle' %name] = np.nan
+          dbData.loc[index, '%s_LineAltDrop' %name] = np.nan
+          distancePath.append(np.nan)
+          distanceEvent.append(np.nan)
+            
     # append to DF
+    #print(distancePath)
     dbData['%s_PathDist' % name] = distancePath
     dbData['%s_LineDist' % name] = distanceEvent
 
     return dbData
+
+
+def findAngleInProfile(pointAngle, avaPath, profile, dsMin):
+
+    anglePara, tmpPara, dsPara = gT.prepareAngleProfile(pointAngle, profile, raiseWarning=False)
+
+    try:
+        indSplitPoint = gT.findAngleProfile(tmpPara, dsPara, dsMin)
+        pointFound = {'x': avaPath['x'][indSplitPoint], 'y': avaPath['y'][indSplitPoint],
+                      'z': avaPath['z'][indSplitPoint], 'zPara': profile['z'][indSplitPoint],
+                      's': profile['s'][indSplitPoint]}
+    except IndexError:
+        noSplitPointFoundMessage = ('Automated split point generation failed as no point where slope is less than %s°'
+                                    'was found, provide the split point manually.' % pointAngle)
+        pointFound = ''
+        #log.warning(noSplitPointFoundMessage)
+
+    return  pointFound
+
+
+def addGradientForPoint (db, pathName, pointList, distance=0.1):
+    """ calculates slope angle 
+
+        Parameters
+        -----------
+       db:
+            dataframe with relevant informations
+       pathName:
+           str with column name of the thalweg
+       pointList:
+           List of str with column names of points to be analysed
+           
+
+        Returns
+        --------
+        db: dataframe with added slope angles
+    """
+    
+    #for loop over point list, to calculate slope angles of all relevant points
+    for attr in pointList:
+        targetPointList = db[attr]
+        db[attr+'_gradient'] = 0
```

Comment:
maybe move this into the loop over the db rows instead of fetching them here into a list and then accessing this list with the index for each row in the next loop like line 231 for point coordinate